docs/documentation/features/frameworkUtilities.md

```diff
@@ -11,4 +11,66 @@ In this chapter useful services/utilities of the Eclipse SmartHome project are d
\ No newline at end of file
+This service is useful for example in the `ThingHandlerFactory` or an `AudioSink` where one needs a specific IP address of the host system to provide something like a `callback` URL.
+
+## Caching
+
+The framework provides some caching solutions for common scenarios.
+
+### Simple expiring and reloading cache
+
+A common usage case is in a `ThingHandler` to encapsulate one value of an internal state and attach an expire time on that value. A cache action will be called to refresh the value if it is expired. This is what `ExpiringCache` implements. If `handleCommand(ChannelUID channelUID, Command command)` is called with the "RefreshType" command, you just return `cache.getValue()`. 
+
+It is a good practice to return as fast as possible from the `handleCommand(ChannelUID channelUID, Command command)` method to not block callers especially UIs.
+Use this type of cache only, if your refresh action is a quick to compute, blocking operation. If you deal with network calls, consider the next cache implementation instead.
+
+### Expiring and asynchronously reloading cache
+
+If we refreshed a value of the internal state in a `ThingHandler` just recently, we can return it immediately via the usual `updateState(channel, state)` method in response to a "RefreshType" command.
+If the state is too old, we need to fetch it first and this may involve network calls, interprocess operations or anything else that will would block for a considerable amout of time.
+
+A common usage case of the `ExpiringCacheAsync` cache type is in a `ThingHandler` to encapsulate one value of an internal state and attach an expire time on that value.
+
+
+A **handleCommand** implementation with the interesting *RefreshType* could look like this:
+```
+public void handleCommand(ChannelUID channelUID, Command command) {
+    if (command instanceof RefreshType) {
+        switch (channelUID.getId()) {
+            case CHANNEL_1:
+                cache1.getValue(updater).thenAccept(value -> updateState(CHANNEL_1, value));
+                break;
+            ...
+        }
+    }
+}
+```
+
+The interesting part is the `updater`. If the value is not yet expired, the returned CompletableFuture will complete immediately and the given code is executed.
+If the value is expired, the updater will be used to request a refreshed value.
+
+An updater can be any class that implements `Supplier<CompletableFuture<VALUE_TYPE>>`. The method `CompletableFuture<VALUE_TYPE> get()` needs to be implemented, for example like this:
+
+```
+class FetchValueFromDevice implements Supplier<CompletableFuture<double>> {
+    CompletableFuture<double> c;
+    
+    @Override
+    CompletableFuture<double> get() {
+       if (c != null) {
+          c = new CompletableFuture<double>();
+          doSuperImportantAsyncStuffHereToGetRefreshedValue();
+       }
+       return c;
+    }
+    
+    @Override
```

Comment:
Right, but then the class should also implement some further custom interface to make clear that this method here is defined in some specific interface (currently, the code suggests that it is a method defined in `Supplier`).